### ISSUE TITLE
UXML, USS Indentation auto formatting and Intellisense

### ISSUE THREAD

**Issue by tsanyqudsi** [microsoft/vscode-dotnettools#325]
### Describe the feature you'd like

As for now, the formatting is using 4 spaces, it would be nice if you could change it to your own preference ( for my case it's 2 ). And it would be nice if you could add intellisense as well.

### Environment Information

- OS: WIndows
- VSCode Version: 1.81.0
-  Extension Version: Unity 0.9.0 (Preview)

**Comment by sailro** [comment]
Could you try to use an `.editorconfig` with the following content ? 

```json
[*.{uss,uxml}]
indent_style = space
indent_size = 2
```

**Comment by tsanyqudsi** [comment]
@sailro does it auto format the file ?

**Comment by sailro** [comment]
Yes, you probably need to install [EditorConfig Extension](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) as well for that.

**Comment by sailro** [comment]
First you need to install the [EditorConfig extension](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig).

Then you need to add ` "editor.formatOnSave": true` in your project `.vscode/settings.json` to enable auto-format on save.

**Comment by tsanyqudsi** [comment]
nope, not working.

**Comment by zoy-l** [comment]
Will uxml/uss code hints be added later? :)

**Comment by PythooonUser** [comment]
I think UXML intellisense has some major challenges involved. As I understand each element in UXML can have custom properties based on the underlying C# class. So the language server for UXML would need to be able to parse C# code first, generate some understanding of what properties exist and then show some completion items, e.g. for built-in classes we would need to read `UnityEngine.UIElements`.

Also, methods like `resource` or `url` in USS need to be able to read your entire `Assets` folder in order to provide completion items for your texture or font assets. As I understand `Resources` folders can appear in any nested folder structure, so we have to keep some kind of index when trying to provide a completion item here.

**Comment by MateuszSadowski** [comment]
I understand that there are challenges involved, but I second the request for Intellisense for UXML.

**Comment by Realinspirer** [comment]
Is this still in the backlog?
